### DISCUSSION TITLE
Migration of a project on Next JS to Nx monorepo.

### DISCUSSION THREAD

**Discussion by YuriyDyachkov** [nrwl/nx#26988] [Feature Requests]
Hello and thanks for your product!
I have a large application created on Next. I want to make this project a monorepository using nx.

How can i do this?

I use the `npx nx@latest init --integrated` command, which I found in the documentation, but I do not get the expected effect with the division of the application into apps and libs as in this [guide](https://nx.dev/recipes/angular/migration/angular#migrating-to-an-integrated-nx-monorepo).

I also encountered a problem that when adding a new application to the nx repository, it installs the latest version of Next, but I need a lower version. How can I use a generator but give it versions?

I apologize if a similar question has been asked before.

**Comment by lamhungypl** [comment]
Any update on this? let's share your jouney @YuriyDyachkov
